Add Txboundary example for Cats Effect (#98)
scalikejdbc/scalikejdbc.github.io: source/documentation/transaction.html.md

## Transaction

<hr/>

> ### #readOnly block / session

Reviewer: please do not change this.

Author: Done.

```

<hr/>

> ### #Working with IO monads

Reviewer: I wanted you to change only this section.

Author: Reverted the titles except this one.

> ### #Working with IO monads

Reviewer: I overlooked this. The `#` in the title is unnecessary
```suggestion
### Working with IO monads
```

Author: Fixed. The other sections had leading `#`, too, and were fixed as well.

```

<hr/>
#### Cats Effect IO example
<hr/>

*TxBoundary typeclass instance for cats.effect.IO[A]*

> `cats.effect.IO` has a cancellation mechanism, which `attempt` can't handle. Here, we use `guarantee`/`guaranteeCase` to make sure the finalizers are run.

Reviewer: ```suggestion
`cats.effect.IO` offers two ways to handle completion/cancellation as below. You can use `guaranteeCase` for commit/rollback operations while going with `guarantee` for connection closure.
```

```scala
import scalikejdbc._
import cats.effect._

implicit def catsEffectIOTxBoundary[A]: TxBoundary[IO[A]] = new TxBoundary[IO[A]] {
  def finishTx(result: IO[A], tx: Tx): IO[A] =

>     result.guaranteeCase{

Reviewer: ```suggestion
    result.guaranteeCase {
```

      case ExitCase.Completed => IO(tx.commit())
      case _ => IO(tx.rollback())
    }

  override def closeConnection(result: IO[A], doClose: () => Unit): IO[A] =
    result.guarantee(IO(doClose()))
}
```

*localTx*

> Since `localTx` performs side-effects such as opening a connection before executing its body, `localTx` itself should be suspended in `IO`. Here, we use `IO.suspend` to achieve this:

Reviewer: ```suggestion
To take control of all the side-effects that happen in `localTx` method, you can use `suspend` method to wrap the code blocks using `localTx`.
```
